src/LoopPartitioningDirective.h

```diff
@@ -0,0 +1,30 @@
+#ifndef HALIDE_LOOP_PARTITIONING_DIRECTIVE_H
+#define HALIDE_LOOP_PARTITIONING_DIRECTIVE_H
+
+/** \file
+ * Defines the LoopPartitionPolicy enum.
+ */
+
+#include <string>
+
+#include "Expr.h"
+#include "Parameter.h"
+
+namespace Halide {
+
+/** Different ways to handle loops with a potentially optimizable boundary conditions. */
+enum class LoopPartitionPolicy {
```

Comment:
I'm going to have to type this a lot (and I regret "TailStrategy" for that reason), so I think this enum class should just be named "Partition"

Reply:
maybe `PartitionPolicy` ?

Reply:
@abadams What about `PartitionPolicy` indeed, with `never_partition()` as sugar?